### DISCUSSION TITLE
Tailwindcss button overriding other plugins css

### DISCUSSION THREAD

**Discussion by t0n1zz** [tailwindlabs/tailwindcss#9800] [Help]
**What version of Tailwind CSS are you using?**

Tailwind 3.2.3

**What version of Node.js are you using?**

-

**What browser are you using?**

brave, chrome

**What operating system are you using?**

macOS 12.6

**Steps to reproduce**

i am using next js + tailwind css and integrating solana wallet adapter that auto generate the button (and css) automatically

i tried at localhost:3000 and everything works just fine, but when i deploy the same code to vercel then tailwind css for button become priority and removing the background color of my solana wallet adapter button

when i do inspect, this styling located on the top 

```
[type=button], [type=reset], [type=submit], button {
    -webkit-appearance: button;
    background-color: transparent;
    background-image: none;
}
```
and overide solana wallet css

```
.wallet-adapter-button-trigger {
    background-color: #512da8;
}
```

for the configuration of my next.js + tailwindcss a plain install from here
```
npx create-next-app --example with-tailwindcss with-tailwindcss-app 
```

and then i only add this to show the solana wallet adapter in` _app.tsx` files

```
import '../styles/globals.css'
import type { AppProps } from 'next/app'
import WalletContextProvider from '../components/WalletContextProvider'

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <WalletContextProvider>
      <Component {...pageProps} />
    </WalletContextProvider>
  )
}

export default MyApp
```

and inside the `WalletContextProvider` is like this

```
require("@solana/wallet-adapter-react-ui/styles.css")
import { FC, ReactNode } from "react"
import {
  ConnectionProvider,
  WalletProvider,
} from "@solana/wallet-adapter-react"
import { WalletModalProvider } from "@solana/wallet-adapter-react-ui"
import { clusterApiUrl } from "@solana/web3.js"
import { PhantomWalletAdapter } from "@solana/wallet-adapter-wallets"
import { useMemo } from "react"

const WalletContextProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const url = useMemo(() => clusterApiUrl("devnet"), [])
  const phantom = new PhantomWalletAdapter()

  return (
    <ConnectionProvider endpoint={url}>
      <WalletProvider wallets={[phantom]} autoConnect>
        <WalletModalProvider>{children}</WalletModalProvider>
      </WalletProvider>
    </ConnectionProvider>
  )
}

export default WalletContextProvider
```

**Comment by wongjn** [comment]
This could be source-order related, whereby the stylesheet for Tailwind CSS is linked after the the solana wallet stylesheet. You could try to resolve this by `@import`ing the sola wallet stylesheet in your Tailwind stylesheet after the `@tailwind base` directive (instead of `import`ing it in JavaScript) to ensure that the solana wallet styles appear later in the source order than the Tailwind preflight styling.

**Reply by adamwathan** [reply]
This is the problem for sure, make sure you are being intentional about your source order, that’s why Tailwind breaks things up into three layers so you have that control 👍🏻

**Reply by t0n1zz** [reply]
so how do i write it in `globals/css` ?
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

what do i need to add if the css supposed to be at `_app.tsx`?
```
import '../styles/globals.css'
import type { AppProps } from 'next/app'
import WalletContextProvider from '../components/WalletContextProvider'

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <WalletContextProvider>
      <Component {...pageProps} />
    </WalletContextProvider>
  )
}

export default MyApp
```

**Reply by t0n1zz** [reply]
that taken from this components

```
require("@solana/wallet-adapter-react-ui/styles.css")
import { FC, ReactNode } from "react"
import {
  ConnectionProvider,
  WalletProvider,
} from "@solana/wallet-adapter-react"
import { WalletModalProvider } from "@solana/wallet-adapter-react-ui"
import { clusterApiUrl } from "@solana/web3.js"
import { PhantomWalletAdapter } from "@solana/wallet-adapter-wallets"
import { useMemo } from "react"

const WalletContextProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const url = useMemo(() => clusterApiUrl("devnet"), [])
  const phantom = new PhantomWalletAdapter()

  return (
    <ConnectionProvider endpoint={url}>
      <WalletProvider wallets={[phantom]} autoConnect>
        <WalletModalProvider>{children}</WalletModalProvider>
      </WalletProvider>
    </ConnectionProvider>
  )
}

export default WalletContextProvider
```

**Reply by wongjn** [reply]
In `WalletContextProvider`:
```diff
-require("@solana/wallet-adapter-react-ui/styles.css")
 import { FC, ReactNode } from "react"
 import {
```

If your build system resolves CSS `@imports`, then in `globals/css`:
```diff
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
 
+@import "@solana/wallet-adapter-react-ui/styles.css";
```

**Reply by t0n1zz** [reply]
not working.. still like this

```
@tailwind base;
@tailwind components;
@tailwind utilities;

@import "@solana/wallet-adapter-react-ui/styles.css";
```

i also try to move it
```
@import "@solana/wallet-adapter-react-ui/styles.css";

@tailwind base;
@tailwind components;
@tailwind utilities;
```

none of them working


<img width="324" alt="Screenshot 2022-11-11 at 18 51 07" src="https://user-images.githubusercontent.com/6964548/201334777-be1060e8-70a2-4dc3-a228-8c9bb322e61a.png">

**Reply by adamwathan** [reply]
Make sure you have `postcss-import` installed and configured and write your CSS this way:

```
@import “tailwindcss/base”;

@import "@solana/wallet-adapter-react-ui/styles.css";

@import “tailwindcss/components”;
@import “tailwindcss/utilities”;
```

That should do it if I’m not making any mistakes, answering from phone so tricky.

**Reply by t0n1zz** [reply]
get this error
```
(21:1) postcss-import: Unable to find uri in '@import “tailwindcss/base”'
```
